tests/run-make/same-lib-two-locations-no-panic/rmake.rs

```diff
@@ -7,22 +7,22 @@
 
 //@ ignore-cross-compile
 
-use run_make_support::{dynamic_lib, rust_lib, rustc, tmp_dir};
+use run_make_support::{dynamic_lib, dynamic_lib_name, rust_lib, rust_lib_name, rustc, tmp_dir};
 use std::fs;
 
 fn main() {
     let tmp_dir_other = tmp_dir().join("other");
 
-    fs::create_dir(&tmp_dir_other);
+    fs::create_dir(&tmp_dir_other).unwrap();
     rustc().input("foo.rs").crate_type("dylib").arg("-Cprefer-dynamic").run();
-    fs::rename(dynamic_lib("foo"), &tmp_dir_other);
+    fs::rename(dynamic_lib("foo"), tmp_dir_other.join(dynamic_lib_name("foo"))).unwrap();
```

Comment:
This seems like it has changed the logic of the test. Was the test broken before?

Reply:
I should've made this into a separate commit, but this test was subtly broken before in that the `fs::rename` actually failed (when I added an `unwrap()`) because the rename `to` is `tmp_dir_other` which is a directory

> On Unix, if `from` is a directory, `to` must also be an (empty) directory. If `from` is not a directory, `to` must also be not a directory. In contrast, on Windows, from can be anything, but to must not be a directory.

In particular, "If `from` is not a directory, `to` must also be not a directory." was violated (`dynamic_lib("foo")` is a file but `tmp_dir_other` is a directory) and produced a panic.

Reply:
Ah, right, I was wondering how it could before, it was swallowing an error. Ok, nice!

Reply:
The test before would actually fail to test the behavior it was trying to test because of the failed rename (it was trying to test same lib in two locations don't cause a panic)

```rs
rustc().input("bar.rs").library_search_path(tmp_dir_other).run();
```

but this line would silently pass even if the compiler did panic because the rename failed.